### PR TITLE
Start tech-preview CSI driver operators when tech preview is enabled

### PR DESCRIPTION
**Pull request by jsafrane**
Start vSphere CSI driver operators only when `TechPreviewNoUpgrade` FeatureSet is enabled.

The CSI driver operator is **not** removed when user removes `TechPreviewNoUpgrade` gate!

cc @openshift/storage

### CODE REVIEW

File: assets/csidriveroperators/vsphere/03_sa.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
```
Comment (jsafrane) [inline assets/csidriveroperators/vsphere/03_sa.yaml:1]: Using `03_sa.yaml` to be able to compare to / copy from the vsphere CSI driver operator `manifests/`

**Review by bertinatto — COMMENTED** [review]
/lgtm

File: Dockerfile.rhel7 [hunk 2]
```diff
@@ -1,9 +1,9 @@
-FROM registry.svc.ci.openshift.org/openshift/release:golang-1.15 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8 AS builder
 WORKDIR /go/src/github.com/openshift/cluster-storage-operator
 COPY . .
 RUN make
 
-FROM registry.svc.ci.openshift.org/openshift/origin-v4.6:base
+FROM registry.ci.openshift.org/ocp/4.8:base
 COPY --from=builder /go/src/github.com/openshift/cluster-storage-operator/cluster-storage-operator /usr/bin/
```
[the remark this answers is not in the document]
Comment (jsafrane) [reply Dockerfile.rhel7:7]: it's used in 'make images', maybe we could delete it.

**Pull request merged**